src/externalConfigFileHandler.ts

```
@@ -0,0 +1,71 @@
+import path from "path"
+import fs from "fs"
+import bytes from "bytes"
+// eslint-disable-next-line import/no-internal-modules
+import compressedSize from "bundlesize/src/compressed-size"
+
+import { BundleSizeConfig } from "./index"
+
+export const createNewConfigFile = (
+  oldConfig: BundleSizeConfig,
+  delta: number,
+  targetSize: number,
+  buildDir: string,
+  fileName?: string
+) => {
+  if (!fileName) {
+    return
+  }
+
+  try {
+    const newConfig = updateConfigurationWithNewBundleSizes(
+      oldConfig,
+      delta,
+      targetSize
+    )
+    fs.writeFileSync(
+      path.join(buildDir, "new-" + fileName),
+      JSON.stringify(newConfig)
+    )
+  } catch (error) {
+    // eslint-disable-next-line no-console
+    console.log("Cannot create new config file", error)
+  }
+}
+
+export const getPreviousConfig = (
+  buildDir: string,
+  fileName?: string
+): BundleSizeConfig | null => {
+  if (!fileName) {
+    return null
+  }
+  try {
+    const config = fs.readFileSync(path.join(buildDir, fileName)).toString()
+    return JSON.parse(config)
+  } catch (error) {
+    // eslint-disable-next-line no-console
+    console.log(
+      "Previous config file not existing or not found... using default values"
```

-> does not exist or invalid json

---

I honestly prefer to throw in unexpected cases instead of falling back. falling back will make the task run through successfully, while it's not at all doing what was intended

---

I don't want to fail the pipeline here. It's fine in my opinion and not a blocker for the pipeline when it fails here. That is why I only log it